src/spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -86,7 +86,23 @@ def _select_extension_data(self, unit_ids):
         new_data = dict(metrics=new_metrics)
         return new_data
 
-    def _run(self, verbose=False, **job_kwargs):
+    def _merge_extension_data(self, units_to_merge, new_unit_ids, new_sorting_analyzer):
+        old_metrics = self.data["metrics"]
+        import pandas as pd
+
+        metrics = pd.DataFrame(index=new_sorting_analyzer.unit_ids)
```

Comment:
If you instantiate this with column names as well, I think you can avoid the loop and set everything with a couple of `loc` calls. I can take a stab at this on Monday